src/HasMedia/HasMediaTrait.php

```diff
@@ -111,6 +114,10 @@ public function addMediaFromUrl(string $url)
         $tmpFile = tempnam(sys_get_temp_dir(), 'media-library');
         file_put_contents($tmpFile, $stream);
 
+        if (! empty($allowedMimeTypes) && Validator::make(['file' => $tmpFile], ['file' => 'mimetypes:'.implode(',', $allowedMimeTypes)])->fails()) {
```

Comment:
Could you put this logic in it's own method? Name it `guardAgainstInvalidMimeType`. Then you can reuse that in `addMediaFromBase64`.

Use early returns in that function instead of using `&&`. So in that function you could do:

```php
if (empty($allowedMimeTypes)) {
   return;
}